Notebooks/Spark.NET C#/GitHub Projects Advanced.ipynb

```diff
@@ -0,0 +1,576 @@
+        "                    \"forked_from INT, deleted STRING, updated_at STRING\")\n",
+        "            .Csv(\"<path_to_ghtorrent/projects.csv>\")\n",
+        "            .Filter(Col(\"language\") == \"C#\");\n",
+        "\n",
+        "projects.Show()"
+      ]
+    },
+    {
+      "cell_type": "markdown",
+      "metadata": {},
+      "source": [
+        "## <font color=purple>Prettier Charting</font> 📃\n",
+        "\n",
+        "That projects DataFrame is pretty intense and not all that easy to read - let's make it much more interesting!\n",
+        "\n",
+        "### We can define a specific formatter that is tied to each DataFrame. \n",
+        "\n",
+        "The next time we display a DataFrame, we'll generate some prettier HTML.\n",
+        "We can make our data match our preferred purple .NET color scheme!"
+      ]
+    },
+    {
+      "cell_type": "code",
+      "execution_count": 6,
+      "outputs": [],
+      "metadata": {
+        "cell_status": {
+          "execute_time": {
+            "duration": 113862.455078125,
+            "end_time": 1571952265574.097
+          }
+        },
+        "collapsed": true
+      },
+      "source": [
+        "Microsoft.DotNet.Interactive.Rendering.Formatter<Microsoft.Spark.Sql.DataFrame>.Register((df, writer) =>\n",
+        "{\n",
+        "    var headers = new List<dynamic>();\n",
+        "    var columnNames = df.Columns();\n",
+        "    headers.Add(th(i(\"index\")));\n",
+        "    headers.AddRange(columnNames.Select(c => th(c)));\n",
+        "\n",
+        "    var rows = new List<List<dynamic>>();\n",
+        "    var currentRow = 0;\n",
+        "    var dfRows = df.Take(20);\n",
+        "    foreach (Row dfRow in dfRows)\n",
+        "    {\n",
+        "        var cells = new List<dynamic>();\n",
+        "        cells.Add(td(currentRow));\n",
+        "\n",
+        "        foreach (string columnName in columnNames)\n",
+        "        {\n",
+        "            cells.Add(td(dfRow.Get(columnName)));\n",
+        "        }\n",
+        "\n",
+        "        rows.Add(cells);\n",
+        "        ++currentRow;\n",
+        "    }\n",
+        "\n",
+        "    var t = table[@border: \"0.1\"](\n",
+        "        thead[@style: \"background-color: purple; color: white; font-family: Verdana\"](headers),\n",
+        "        tbody[@style: \"color: indigo; font-size: 14px\"](rows.Select(r => tr(r))));\n",
+        "\n",
+        "    writer.Write(t);\n",
+        "}, \"text/html\");"
+      ]
+    },
+    {
+      "cell_type": "markdown",
+      "metadata": {},
+      "source": [
+        "### Let's display a prettier version of our projects DataFrame.\n",
+        "We can display a more concise subset of the data and use our beautiful new HTML rendering."
+      ]
+    },
+    {
+      "cell_type": "code",
+      "execution_count": 7,
+      "outputs": [
+        {
+          "output_type": "execute_result",
```

Comment:
When I ran this cell, I got:

```Error : (1,1): error CS0234: The type or namespace name 'Rendering' does not exist in the namespace 'Microsoft.DotNet.Interactive' (are you missing an assembly reference?)```

Reply:
This still works when I run it in Synapse today, but maybe it doesn't work in the latest version of Synapse (that hasn't been merged into the prod version I'm using to test)? I suspect that could be the case also because I googled `Microsoft.DotNet.Interactive.Rendering.Formatter`, and the link that came up is now broken (maybe it is out of date in the latest version of .NET interactive now?): https://github.com/dotnet/try/tree/master/Microsoft.DotNet.Interactive.Formatting